src/services/news/index.js

```diff
@@ -1,23 +1,233 @@
+			if (scopeListService === undefined) {
+				throw new Error(`Missing ScopeListService for scope "${scope}".`);
+			}
+			let scopeItems = await scopeListService.find({
+				route: { scopeId: userId.toString() },
+				query: { permissions: ['NEWS_VIEW'] },
+			});
+			if (target) {
+				// if a target id is given, only return news from this target
+				scopeItems = scopeItems.filter(i => i._id.toString() === target.toString());
+			}
+			return Promise.all(scopeItems.map(async (item) => {
+				const news = await newsModel.find({
+					targetModel: scope,
+					target: item._id,
+				}).lean();
+				// Manually populate the target (current API requires this):
+				return Promise.all(news.map(async n => ({
+					...n,
+					target: await this.app.service(scope).get(n.target),
+				})));
+			}));
+		});
+		const results = await Promise.all(ops);
+		return flatten(results);
+	}
+
+	/**
+	 * GET /news/{id}
+	 * Returns the news item specified by id
+	 * @param {BsonId|String} id
+	 * @param {Object} params
+	 * @returns one news item
+	 * @throws {Forbidden} if not authorized
+	 * @throws {NotFound} if the id does not belong to a news object
+	 * @memberof NewsService
+	 */
+	async get(id, params) {
+		const query = { _id: id };
+		const news = await newsModel.findOne(query).lean();
+		this.checkExistence(news, query);
+		await this.authorize(news, params.account, 'NEWS_VIEW');
+		return news;
+	}
+
+	/**
+	 * GET /news/
+	 * Returns all news the user can see.
+	 * @param {Object} params
+	 * @returns paginated array of news items
+	 * @memberof NewsService
+	 */
+	async find(params) {
+		let news = [];
+		const scoped = params.query && params.query.target;
+		if (scoped) {
+			news = news.concat(await this.findScopedNews(params.account.userId, params.query.target));
+		} else {
+			news = news.concat(await this.findSchoolNews(params.account));
+			news = news.concat(await this.findScopedNews(params.account.userId));
+		}
+		return Promise.resolve(paginate(news, params));
+	}
+
+	/**
+	 * POST /news/
+	 * Creates a news item
+	 * @param {Object} data @see NewsModel
+	 * @param {Object} params
+	 * @returns {News} the created news object
+	 * @throws {Forbidden} if not authorized
+	 * @memberof NewsService
+	 */
+	async create(data, params) {
+		await this.authorize(data, params.account, 'NEWS_CREATE');
+		return newsModel.create(data);
+	}
+
+	/**
+	 * DELETE /news/{id}
+	 * Deletes a single news item.
+	 * @param {BsonId|String} id The news item's Id
+	 * @param {Object} params Note that params.query won't work here
+	 * @returns {News} the deleted news item
+	 * @throws {Forbidden} if not authorized
+	 * @throws {NotFound} if the id does not belong to a news object
+	 * @memberof NewsService
+	 */
+	async remove(id, params) {
+		const query = { _id: id };
+		const news = await newsModel.findOne(query).lean();
+		this.checkExistence(news, query);
+		await this.authorize(news, params.account, 'NEWS_CREATE');
+		await newsModel.findByIdAndDelete(id);
```

Comment:
Consider a soft delete here or at least store the old news somewhere.. same for updates ...

Reply:
For updates/patches, a NewsHistory document is created by the service. On delete, all NewsHistory documents are deleted with the News document. I think this needs to be discussed.